test/com/linkedin/drelephant/spark/fetchers/SparkRestClientTest.scala

```diff
@@ -70,12 +74,18 @@ class SparkRestClientTest extends AsyncFunSpec with Matchers {
       sparkRestClient.fetchData(FetchClusterModeDataFixtures.APP_ID) map { restDerivedData =>
         restDerivedData.applicationInfo.id should be(FetchClusterModeDataFixtures.APP_ID)
         restDerivedData.applicationInfo.name should be(FetchClusterModeDataFixtures.APP_NAME)
-        restDerivedData.jobDatas should not be(None)
-        restDerivedData.stageDatas should not be(None)
-        restDerivedData.executorSummaries should not be(None)
+        restDerivedData.jobDatas should not be (None)
+        restDerivedData.stageDatas should not be (None)
+        restDerivedData.executorSummaries should not be (None)
+        restDerivedData.logDerivedData should be(None)
+      } flatMap {
+        case assertion: Try[Assertion] => assertion
+        case _ =>
+          sparkRestClient.fetchData(FetchClusterModeDataFixtures.APP_ID, fetchLogs = true)
+            .map { _.logDerivedData should not be(None) }
```

Comment:
I agree that SparkLogClientTest is already testing the parsing, but in SparkRestClient, you are using a different logic to get the file. The only way you can test that your logic is correctly reading the file without corrupting the file is by validating the content of the file.